packages/lodestar/src/chain/blocks/process.ts

```diff
@@ -40,7 +43,13 @@ export async function processBlock(modules: BlockProcessModules, job: IBlockJob)
     });
   }
 
-  await processBlocksInEpoch(modules, job, [job.signedBlock]);
+  const {signedBlock} = job;
+  await processBlocksInEpoch(modules, job, [signedBlock]);
+  if (metrics) {
+    // Returns the delay between the start of `block.slot` and `current time`
+    const delaySec = Date.now() / 1000 - (genesisTime + signedBlock.message.slot * config.SECONDS_PER_SLOT);
+    metrics.gossipBlock.elappsedTimeTillProcessed.observe(delaySec);
```

Comment:
Note sync and API submited blocks also come through this method so they will distort the metrics

Reply:
Oh, now that I just merge the unknown sync PR you can just use the processBlock promise in the gossip handler to time when it's done processing